solr/core/src/java/org/apache/solr/update/processor/DistributedZkUpdateProcessor.java

```diff
@@ -184,6 +186,30 @@ public void processCommit(CommitUpdateCommand cmd) throws IOException {
 
     updateCommand = cmd;
 
+    // 1. SHARED replica has a hard requirement of processing each indexing batch with a hard commit(either explicit or
+    // implicit, HttpSolrCall#addCommitIfAbsent) because that is how, at the end of an indexing batch, synchronous push
+    // to shared store gets hold of the segment files on local disk. SHARED replica also does not support the notion of soft commit.
+    // Therefore unlike NRT replica type we do not need to broadcast commit to the leaders of all the shards of a collection.
```

Comment:
Correct. Former is not needed because isolated commit is a no-op for SHARED replica and later is not supported because SHARED replica has a different plan around opening of searchers https://issues.apache.org/jira/browse/SOLR-14339
I have updated the comments.